tests/test_cleaning.py

```diff
@@ -0,0 +1,204 @@
+import random
+import pytest
+from copy import deepcopy
+
+BANK_PROCESSING_TIME_IN_DAYS = 5
+
+INTERNAL_TRANSFER_REFERENCES = [
+    "internal reference 0",
+    "internal reference 1",
+]
+
+REGULAR_TRANSACTIONS_LIST = [
+    SimpleTransaction(
+        transactionId="transaction_dummy_1",
+        datetime=datetime.now(),
+        amount=0.1,
+        referenceText="reference_transaction_dummy_1",
+        bankTransactionCode="dummy_transaction_code",
+    ),
+    SimpleTransaction(
+        transactionId="transaction_dummy_2",
+        datetime=datetime.now(),
+        amount=0.2,
+        referenceText="reference_transaction_dummy_2",
+        bankTransactionCode="dummy_transaction_code",
+    ),
+]
+
+
+@pytest.fixture(autouse=True)
+def user_config_mock() -> Generator[Mock, None, None]:
+    with patch(
+        "personal_finances.transaction.cleaning.get_user_configuration"
+    ) as u_mock:
+        u_mock.return_value.InternalTransferReferences = INTERNAL_TRANSFER_REFERENCES
+        u_mock.return_value.BankProcessingTimeInDays = BANK_PROCESSING_TIME_IN_DAYS
+        yield u_mock
+
+
+def create_pair_internal_transaction(
+    amount: Optional[float] = None,
+) -> List[SimpleTransaction]:
+    transfer_amount = random.random() if amount is None else amount
+    transfer_datetime = datetime.now()
+
+    internal_trasaction_pair = [
+        simple_transaction_builder(
+            transactionId=str(random.random()),
+            datetime=transfer_datetime,
+            amount=transfer_amount,
+            referenceText="dummy noise"
+            + INTERNAL_TRANSFER_REFERENCES[0]
+            + "data dummy data dummy",
+            bankTransactionCode="dummy_transaction_code",
+        ),
+        simple_transaction_builder(
+            transactionId=str(random.random()),
+            datetime=transfer_datetime,
+            amount=-transfer_amount,
+            referenceText="dummy noise"
+            + INTERNAL_TRANSFER_REFERENCES[1]
+            + "data dummy data dummy",
+            bankTransactionCode="dummy_transaction_code",
+        ),
+    ]
+    return internal_trasaction_pair
+
+
+def create_list_with_internal_transactions(
+    amount: Optional[float] = None,
+) -> List[SimpleTransaction]:
+    return (
+        create_pair_internal_transaction(amount)
+        + create_pair_internal_transaction(amount)
+        + create_pair_internal_transaction(amount)
+    )
+
+
+def copy_first_element(
+    internal_transfer_pair: List[SimpleTransaction],
+) -> SimpleTransaction:
+
+    duplicated_transfer_element = deepcopy(internal_transfer_pair[0])
+    duplicated_transfer_element["transactionId"] = str(random.random())
+
+    return duplicated_transfer_element
+
+
+def assert_transaction_list(
+    input_transaction_list: List[SimpleTransaction],
+    expected_cleaned_list: List[SimpleTransaction],
+) -> None:
+    assert remove_internal_transfers(input_transaction_list) == expected_cleaned_list
+
```

Comment:
I'm missing a "empty list" test.

Reply:
Good point. I've wiped it out by mistake